pkg/bench/bench.go

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package bench
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"path"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/cilium/tetragon/api/v1/tetragon"
+	"github.com/cilium/tetragon/pkg/api/readyapi"
+	"github.com/cilium/tetragon/pkg/bpf"
+	"github.com/cilium/tetragon/pkg/btf"
+	"github.com/cilium/tetragon/pkg/cilium"
+	"github.com/cilium/tetragon/pkg/config"
+	"github.com/cilium/tetragon/pkg/exporter"
+	"github.com/cilium/tetragon/pkg/grpc"
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/cilium/tetragon/pkg/observer"
+	"github.com/cilium/tetragon/pkg/option"
+	"github.com/cilium/tetragon/pkg/process"
+	"github.com/cilium/tetragon/pkg/reader/notify"
+	"github.com/cilium/tetragon/pkg/sensors"
+	"github.com/cilium/tetragon/pkg/sensors/base"
+	"github.com/cilium/tetragon/pkg/watcher"
+
+	// Imported to allow sensors to be initialized inside init().
+	_ "github.com/cilium/tetragon/pkg/sensors/exec"
+	_ "github.com/cilium/tetragon/pkg/sensors/tracing"
+)
+
+type Arguments struct {
+	Trace       string
+	Crd         string
+	CmdArgs     []string
+	Debug       bool
+	PrintEvents bool
+	JSONEncode  bool
+	Baseline    bool
+}
+
+func readConfig(file string) (*config.GenericTracingConf, error) {
+	if file == "" {
+		return nil, nil
+	}
+
+	yamlData, err := os.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read yaml file %s: %w", file, err)
+	}
+	cnf, err := config.ReadConfigYaml(string(yamlData))
+	if err != nil {
+		return nil, err
+	}
+
+	return cnf, nil
+}
+
+func (args *Arguments) String() string {
+	return fmt.Sprintf("")
+}
+
+func runTetragon(ctx context.Context, configFile string, args *Arguments, summary *Summary, ready chan bool) {
+	bpf.ConfigureResourceLimits()
+	bpf.CheckOrMountFS("")
+	bpf.CheckOrMountDebugFS()
+	bpf.CheckOrMountCgroup2()
+
+	if args.Debug {
+		option.Config.Verbosity = 5
+	}
+
+	if _, err := os.Stat("../../bpf/objs"); err == nil {
+		option.Config.HubbleLib = "../../bpf/objs"
+	} else {
+		exePath, err := os.Executable()
+		if err != nil {
+			log.Fatal(err)
+		}
+		option.Config.HubbleLib = path.Join(path.Dir(exePath), "bpf/objs")
+
+		if _, err := os.Stat(option.Config.HubbleLib); err != nil {
+			// Running outside the source tree, fall back to default location.
+			option.Config.HubbleLib = "/var/lib/hubble-fgs"
```

Comment:
Maybe this one: https://github.com/cilium/tetragon/blob/main/pkg/defaults/defaults.go#L44 ?

Reply:
ok, will use that one, thanks